src/PIL/ImageCms.py

```diff
@@ -576,15 +582,15 @@ def buildTransform(
 
 
 def buildProofTransform(
-    inputProfile,
-    outputProfile,
-    proofProfile,
-    inMode,
-    outMode,
-    renderingIntent=Intent.PERCEPTUAL,
-    proofRenderingIntent=Intent.ABSOLUTE_COLORIMETRIC,
-    flags=Flags.SOFTPROOFING,
-):
+    inputProfile: _CmsProfileCompatible,
+    outputProfile: _CmsProfileCompatible,
+    proofProfile: _CmsProfileCompatible,
+    inMode: str,
+    outMode: str,
+    renderingIntent: Intent = Intent.PERCEPTUAL,
+    proofRenderingIntent: Intent = Intent.ABSOLUTE_COLORIMETRIC,
+    flags: Flags | int = Flags.SOFTPROOFING,
```

Comment:
Wouldn't our discussion in https://github.com/python-pillow/Pillow/pull/7671#discussion_r1438996324 about favouring enums over `int` also apply to the `IntFlag` of `Flags`, meaning this could just be `flags: Flags`?

Reply:
Yep, you're right. IIRC I might have added `int` to support the deprecated `ImageCms.FLAGS`, but that does not have type hints and should not be used anyway.